### PR TITLE
Feature: support http

### LINKED ISSUE #11: Meet certificate verify failed when I scan my private image registry
when I scan my private image registry, it meet the SSL error.
I can not find the solution ,anyone can help me? Thanks

[root@mirror anchors]# claircli -c http://mirror.redhat.ren:6060  mirror.redhat.ren/admin/tomcat:latest
2020-01-27 15:49:26,212|INFO|*****************************1******************************
2020-01-27 15:49:26,212|INFO|Analyzing <Image: mirror.redhat.ren/admin/tomcat:latest>
2020-01-27 15:49:26,229|WARNING|HTTPSConnectionPool(host='mirror.redhat.ren', port=443): Max retries exceeded with url: /v2/ (Caused by SSLError(SSLError(1, u'[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:618)'),))
2020-01-27 15:49:26,229|INFO|============================================================
2020-01-27 15:49:26,229|INFO|            CLAIR ANALYSIS REPORTS: (1) IN TOTAL            
2020-01-27 15:49:26,229|INFO|============================================================
2020-01-27 15:49:26,229|ERROR|IMAGES WERE ANALYZED WITH ERROR (1)
2020-01-27 15:49:26,229|ERROR|mirror.redhat.ren/admin/tomcat:latest

### PR DESCRIPTION
**Pull request by toydarian**
I added a flag to specify the scheme (http/https) to access the docker registry as https was hardcoded and I needed http for a test with a temporary registry and I didn't want to setup SSL for it.

### CODE REVIEW

**Comment by joelee2012** [comment]
thanks for opening pull request! however i perfer to add an appendable argument --insecure-registry, this list can be set as class variable of RemoteRegistry, any domain in this list should use http schema. with this you can mix use insecure/secure registry.

**Comment by toydarian** [comment]
don't merge, it has a bug

**Comment by toydarian** [comment]
@joelee2012 if we want to have a list with insecure registries, we have the following problem:  
If we run `claircli --insecure-registries localhost:5000 some-image:tag` argparse will think that `some-image:tag` belongs to the list of insecure registries and will fail because it can't find the positional argument.  
The only way around, that I currently see, would be to make the images a flagged argument like this: `claircli --insecure-registries localhost:5000 -i some-image:tag`, but that would break backwards compatibility.

**Comment by joelee2012** [comment]
the '--insecure-registry' should be an argument with option 'action=append' like
`parser.add_argument('-i', '--insecure-registry', action='append', help='name of insecure registry')`

**Comment by toydarian** [comment]
I did exactly that, but while writing tests, I found out, that when you run the command `claircli --insecure-registries localhost:5000 some-image:tag` it will not recognize the last argument as a positional argument, but assume it belongs to the list of insecure registries which will lead to an error because it misses the required positional argument.

**Comment by joelee2012** [comment]
no, you don't , you use 'nargs＝＋' instead of 'action＝append'

**Comment by toydarian** [comment]
Ah, I learnt something today, thanks!  
I made the changes and rebased the branch to remove the clutter.
imho you can merge

File: claircli/cli.py [hunk 1]
```diff
@@ -84,6 +88,10 @@ def __init__(self):
         parser.add_argument('-L', '--log-file', help='save log to file')
         parser.add_argument(
             '-d', '--debug', action='store_true', help='print more logs')
+        parser.add_argument('-i', '--insecure-registries',
+                            action='append', metavar='some.reg:port',
```
Comment (joelee2012) [inline claircli/cli.py:92]: should be --insecure-registry and metavar should be registry

File: claircli/cli.py [hunk 2]
```diff
@@ -96,6 +104,10 @@ def __init__(self):
         parser.set_defaults(func=self.analyze_image)
         self.args = parser.parse_args()
         self.setup_logging()
+        if isinstance(self.args.insecure_registries, Iterable):
+            self.insec_regs = frozenset(self.args.insecure_registries)
```
Comment (joelee2012) [inline claircli/cli.py:108]: the insecure_registries is a list which is iterable. just covert to set

File: claircli/cli.py [hunk 3]
```diff
@@ -118,7 +130,7 @@ def resolve_images(self, images):
         result = set()
         for pattern in images:
             reg, repo, tag = Image.parse_id(pattern)
-            registry = RemoteRegistry(reg)
+            registry = RemoteRegistry(reg, insecure_registries=self.insec_regs)
```
Comment (joelee2012) [inline claircli/cli.py:133]: the list should be class variable instead of parameter of construct.

**Comment by joelee2012** [comment]
i will implement this feature in #13

**Pull request closed without merging**